cherrypy/_cprequest.py

```diff
@@ -99,19 +99,19 @@ def run(self, point):
             # but proceed on to the next hook. The only way
             # to stop all processing from one of these hooks is
             # to raise SystemExit and stop the whole server.
-            if exc is None or hook.failsafe:
+            if not raising or hook.failsafe:
                 try:
                     hook()
                 except (KeyboardInterrupt, SystemExit):
                     raise
                 except (cherrypy.HTTPError, cherrypy.HTTPRedirect,
                         cherrypy.InternalRedirect):
-                    exc = sys.exc_info()[1]
+                    raising = True
                 except Exception:
-                    exc = sys.exc_info()[1]
+                    raising = True
                     cherrypy.log(traceback=True, severity=40)
-        if exc:
-            raise exc
+        if raising:
+            raise
```

Comment:
I'm pretty sure this will not work because there's nothing to raise in this context (outside of the except block)...